src/Gateway.php

```diff
@@ -164,6 +164,54 @@ public function updateSubscription(array $parameters = array())
         return $this->createRequest('\Omnipay\Vindicia\Message\CreateSubscriptionRequest', $parameters, true);
     }
 
+    /**
+     * Fetch the invoice number of a subscription
+     *
+     * See Message\FetchSubscriptionInvoiceRequest for more details.
+     *
+     * @param array $parameters
+     * @return \Omnipay\Vindicia\Message\FetchSubscriptionInvoiceRequest
+     */
+    public function fetchSubscriptionInvoiceNumbers(array $parameters = array())
+    {
+        /**
+         * @var \Omnipay\Vindicia\Message\FetchSubscriptionInvoiceRequest
+         */
+        return $this->createRequest('\Omnipay\Vindicia\Message\FetchSubscriptionInvoiceRequest', $parameters);
+    }
+
+    /**
+     * Fetch the invoice of a subscription given invoice number
+     *
+     * See Message\FetchSubscriptionInvoiceRequest for more details.
+     *
+     * @param array $parameters
+     * @return \Omnipay\Vindicia\Message\FetchSubscriptionInvoiceRequest
+     */
+    public function fetchSubscriptionInvoice(array $parameters = array())
+    {
+        /**
+         * @var \Omnipay\Vindicia\Message\FetchSubscriptionInvoiceRequest
+         */
+        return $this->createRequest('\Omnipay\Vindicia\Message\FetchSubscriptionInvoiceRequest', $parameters);
+    }
+
+    /**
+     * Make payment to a specific amount or an outstanding invoice for a subscription
+     *
+     * See Message\MakePaymentRequest for more details.
+     *
+     * @param array $parameters
+     * @return \Omnipay\Vindicia\Message\MakePaymentRequest
+     */
+    public function makePayment(array $parameters = array())
```

Comment:
here's an idea, and i don't know if it's good: we could call this `charge`. That might make it more clear that a user is actually getting charged. I believe makePayment has the ability to record a payment without actually charging the user too, though. But maybe if we ever supported that, we would call it recordPayment? just trying to think of what naming makes most sense from an Omnipay perspective.

cc @parth21shah

Reply:
I would go with a charge. I think recordPayment is just for recording purpose it infers payment has ben made we are just recording it

Reply:
I think we should still go with makePayment, since makePayment also indicates the `charge` action to clear an outstanding subscription invoice balance. v18 also has an customer::makePayment methods to pay off balance for an account. I think unless other gateways call it `charge` and you want to unify the naming, I vote for keeping the name as it is.

Reply:
i am trying to think of what other omnipay gateways would call it. unfortunately there's not a great way to know... could you take a peek at a few and see if you see anything?

i suspect they wouldn't call it makePayment because "payment" isn't really a word omnipay uses. but then again, idk if other gateways support this at all.

we could do something like makeTransaction... this does generate a transaction record, right?

Reply:
talked with nicky offline and we think it's ok to keep the name as this function is really vindicia specific and doesn't get used in other omnipay libraries.